File: Core/NakedObjects.Persistor.Entity.Test.Domain/ProductProductPhoto.cs
using System;
using NakedObjects;

namespace AdventureWorksModel
{
    public partial class ProductProductPhoto
    {
    
        #region Primitive Properties
        #region ProductID (Int32)
    [MemberOrder(100)]
        public virtual int  ProductID {get; set;}

        #endregion

        #region ProductPhotoID (Int32)
    [MemberOrder(110)]
        public virtual int  ProductPhotoID {get; set;}

        #endregion

        #region Primary (Boolean)
    [MemberOrder(120)]
        public virtual bool  Primary {get; set;}

        #endregion

        #region ModifiedDate (DateTime)
    [MemberOrder(130), Mask("d")]
        public virtual DateTime  ModifiedDate {get; set;}

        #endregion


        #endregion

        #region Navigation Properties
        #region Product (Product)
    		
    [MemberOrder(140)]
    	public virtual Product Product {get; set;}

        #endregion

        #region ProductPhoto (ProductPhoto)
    		
    [MemberOrder(150)]
    	public virtual ProductPhoto ProductPhoto {get; set;}

        #endregion


        #endregion

    }
}

File: Test/AdventureWorksLegacyModelVB/Production/ProductProductPhoto.vb
Namespace AW.Types

	Partial Public Class ProductProductPhoto
 Implements ITitledObject
		Public Property ProductID() As Integer

		Public Property ProductPhotoID() As Integer

#Region "Primary"
        Public mappedPrimary As Boolean
        Friend myPrimary As Logical

        <MemberOrder(1)>
        Public ReadOnly Property Primary As Logical
            Get
                Return If(myPrimary, New Logical(mappedPrimary, Function(v) mappedPrimary = v))
            End Get
        End Property

        Public Sub AboutPrimary(a As FieldAbout, Primary As Logical)
            Select Case a.TypeCode
                Case AboutTypeCodes.Name
                Case AboutTypeCodes.Usable
                Case AboutTypeCodes.Valid
                Case AboutTypeCodes.Visible
            End Select
        End Sub
#End Region

        Public Overridable Property Product() As Product

        Public Overridable Property ProductPhoto() As ProductPhoto

#Region "ModifiedDate"
		Public mappedModifiedDate As Date
		Friend myModifiedDate As TimeStamp

		<MemberOrder(99)>
		Public ReadOnly Property ModifiedDate As TimeStamp
			Get
				Return If(myModifiedDate, New TimeStamp(mappedModifiedDate, Function(v) mappedModifiedDate = v))
			End Get
		End Property

		Public Sub AboutModifiedDate(a As FieldAbout, ModifiedDate As TimeStamp)
			Select Case a.TypeCode
				Case AboutTypeCodes.Usable
					a.Usable = False
			End Select
		End Sub
#End Region

        Public Function Title() As ITitle Implements ITitledObject.Title
            Return New Title(ToString())
        End Function

        Public Overrides Function ToString() As String
            Return $"ProductProductPhoto: {ProductID}-{ProductPhotoID}"
        End Function
    End Class
End Namespace